### PR TITLE
Add Fleet and Elastic Agent 8.4 release notes

### PR DESCRIPTION
**Pull request by dedemorton**
Adds release notes for Elastic Agent and Fleet in 8.4 based on changelog entries.

@jen-huang 
* I've edited the list of Fleet items taken from https://github.com/elastic/kibana/pull/137621. Some entries were confusing and  inconsistent. Please review, and I will make sure the changes get back into the Kibana docs.
* "Enables package signature verification feature"  shows up in the 8.4 release notes, but I thought we were going to wait and not expose this feature in 8.4 GA.

@jlind23 

Several of the Elastic Agent bug fixes were in 8.3.3, but since they showed up in the changelog and commit history for 8.4.0, I've included them. They are the first 6 in the list. Let me know if I should keep or remove them.

We need to get this merged tomorrow. 

TODO after merging:
- [ ] Edit Kibana release notes.
- [ ] Delete 8.3 release notes file from main.

### CODE REVIEW

**Review by jlind23 — APPROVED** [review]
Looks good to me, thanks @dedemorton 

cc @pierrehilbert for awareness

File: docs/en/ingest-management/release-notes/release-notes-8.4.asciidoc [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+[discrete]
+[[new-features-8.4.0]]
+=== New features
+
+The 8.4.0 release adds the following new and notable features. 
+ 
+{fleet}::
+* Enable package signature verification feature {kibana-pull}137239[#137239]
+* Allow user to force install an unverified package {kibana-pull}136108[#136108]
+* Display package verification status {kibana-pull}135928[#135928]
```
Comment (jen-huang) [inline docs/en/ingest-management/release-notes/release-notes-8.4.asciidoc:61]: Yeah, let's skip the release notes for the package verification items until it is turned on

**Review by jen-huang — APPROVED** [review]

File: docs/en/ingest-management/release-notes/release-notes-8.4.asciidoc (hunk 1 above)
Comment (dedemorton) [inline docs/en/ingest-management/release-notes/release-notes-8.4.asciidoc:61]:
```suggestion
```

File: docs/en/ingest-management/release-notes/release-notes-8.4.asciidoc [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+====
+
+[discrete]
+[[new-features-8.4.0]]
+=== New features
+
+The 8.4.0 release adds the following new and notable features. 
+ 
+{fleet}::
+* Enable package signature verification feature {kibana-pull}137239[#137239]
```
Comment (dedemorton) [inline docs/en/ingest-management/release-notes/release-notes-8.4.asciidoc:59]:
```suggestion
```

**Pull request merged**